opentelemetry-sdk/tests/metrics/integration_test/test_delta.py

```diff
@@ -0,0 +1,130 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from io import StringIO
+from itertools import count
+from json import loads
+from time import sleep
+from unittest import TestCase
+
+from opentelemetry.metrics import (
+    Observation,
+    get_meter_provider,
+    set_meter_provider,
+)
+from opentelemetry.sdk.metrics import MeterProvider, ObservableCounter
+from opentelemetry.sdk.metrics.export import (
+    AggregationTemporality,
+    ConsoleMetricExporter,
+    PeriodicExportingMetricReader,
+)
+from opentelemetry.sdk.metrics.view import SumAggregation
+from opentelemetry.test.globals_test import reset_metrics_globals
+
+network_bytes_generator = count(start=8, step=8)
+
+
+def observable_counter_callback(callback_options):
+
+    yield Observation(next(network_bytes_generator))
+
+
+class TestDelta(TestCase):
+    def test_observable_counter_delta(self):
+        def setUp(self):
+            reset_metrics_globals()
+
+        def tearDown(self):
+            reset_metrics_globals()
+
+        output = StringIO()
+
+        aggregation = SumAggregation()
+
+        exporter = ConsoleMetricExporter(
+            out=output,
+            preferred_aggregation={ObservableCounter: aggregation},
+            preferred_temporality={
+                ObservableCounter: AggregationTemporality.DELTA
+            },
+        )
+
+        reader = PeriodicExportingMetricReader(
+            exporter, export_interval_millis=100
+        )
+
+        provider = MeterProvider(metric_readers=[reader])
+        set_meter_provider(provider)
+
+        meter = get_meter_provider().get_meter(
+            "preferred-aggregation", "0.1.2"
+        )
+
+        meter.create_observable_counter(
+            "observable_counter", [observable_counter_callback]
+        )
+
+        sleep(1)
+
+        provider.shutdown()
+
+        output.seek(0)
+
+        joined_output = "".join(output.readlines())
+
+        stack = []
```

Comment:
This is pretty awkward. Maybe we should have an exporter that gives us easier to parse data, even if it is only for testing purposes.

Reply:
Probably should use `InMemoryMetricReader`.